x/posts/internal/types/msgs.go

```diff
@@ -94,15 +116,65 @@ func (msg MsgCreatePost) ValidateBasic() sdk.Error {
 }
 
 // GetSignBytes encodes the message for signing
-func (msg MsgCreatePost) GetSignBytes() []byte {
+func (msg MsgCreateTextPost) GetSignBytes() []byte {
 	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(msg))
 }
 
 // GetSigners defines whose signature is required
-func (msg MsgCreatePost) GetSigners() []sdk.AccAddress {
+func (msg MsgCreateTextPost) GetSigners() []sdk.AccAddress {
 	return []sdk.AccAddress{msg.Creator}
 }
 
+// ----------------------
+// --- MsgCreateMediaPost
+// ----------------------
+
+// MsgCreateMediaPost defines a CreateMediaPost message
+type MsgCreateMediaPost struct {
```

Comment:
Do we really want a new type just to add medias? 

Maybe we should go with encapsulations instead, transforming the original `MsgCreatePost` into 


```golang
type MsgCreatePost struct {
  // All post fields
  Medias        PostMedias        `json:"post_medias,omitempty"`
}
```

We also should have a new method called `NewMsgCreatePostWithMedia`: 

```golang
func NewMsgCreatePostWithMedia(<all post data>, medias PostMedias) MsgCreatePost {
  // ...
}
```